packages/docs/community/release-process/smart-contracts.md

```diff
@@ -72,11 +72,23 @@ yarn verify-deployed -n $NETWORK -b $PREVIOUS_RELEASE -f
 
 ### Check Backward Compatibility
 
-Use the following script to compile candidate and current release contracts to check storage layout and ABI backwards compatibility, subject to the following exceptions:
+This script performs some automatic checks to ensure that smart contract
+versions have been updated correctly.
+
+Specifically, it compiles candidate and current release contracts to check storage layout and ABI backwards compatibility, subject to the following exceptions:
   1. If the STORAGE version has changed, does not perform backwards compatibility checks
   2. If the MAJOR version has changed, checks storage layout compatibility but not ABI compatibility
 
-For changed contracts, checks that new version number is strictly greater than previous release. For unchanged contracts, checks that version number exactly matches previous release. Outputs a JSON backwards compatibility report.
+Versions being updated correctly means that no backwards incompatibilities were
+introduced on accident. For changed contracts, checks that new version number is strictly greater than previous release. For unchanged contracts, checks that version number exactly matches previous release.
+​
+Critically, this ensures that proxied contracts do not experience storage
+collisions between implementation versions. See [this
+article](https://docs.openzeppelin.com/upgrades-plugins/1.x/proxies#storage-collisions-between-implementation-versions)
+by OpenZeppelin for a good overview of this problem and why it's important to
+check for it.
+​
+The script generates a detailed report on version changes in JSON format.
```

Comment:
line 105 reads: `These new contracts may need to be initialized, for which values can be provided. Outputs a JSON contract upgrade governance proposal.` in reference to : `releaseData/initializationData/release${N}.json` 
The second sentence is not a complete sentence and in fact, perhaps a typo. I don't think we ever explain where that JSON file came from (or was it constructed manually?)

Reply:
Yeah it is in reference to the releaseData which is checked-in. As a deployer, you can probably just use that, as a verifier, in the later step, one should take it from the CGP (and I hope that this changes makes that more clear)